exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverExporterWorker.java

```diff
@@ -93,27 +104,44 @@ boolean registerView(View view) {
     }
     registeredViews.put(view.getName(), view);
 
-    // TODO(songya): don't need to create MetricDescriptor for RpcViewConstants once we defined
-    // canonical metrics. Registration is required only for custom view definitions. Canonical
-    // views should be pre-registered.
-    MetricDescriptor metricDescriptor =
-        StackdriverExportUtils.createMetricDescriptor(view, projectId);
-    if (metricDescriptor == null) {
-      // Don't register interval views in this version.
-      return false;
-    }
+    Span span = tracer.getCurrentSpan();
+    span.addAnnotation("Create Stackdriver Metric.");
+    try (Scope scope = tracer.withSpan(span)) {
+      // TODO(songya): don't need to create MetricDescriptor for RpcViewConstants once we defined
+      // canonical metrics. Registration is required only for custom view definitions. Canonical
+      // views should be pre-registered.
+      MetricDescriptor metricDescriptor =
+          StackdriverExportUtils.createMetricDescriptor(view, projectId);
+      if (metricDescriptor == null) {
+        // Don't register interval views in this version.
+        return false;
+      }
 
-    CreateMetricDescriptorRequest request =
-        CreateMetricDescriptorRequest.newBuilder()
-            .setNameWithProjectName(projectName)
-            .setMetricDescriptor(metricDescriptor)
-            .build();
-    try {
-      metricServiceClient.createMetricDescriptor(request);
-      return true;
-    } catch (Throwable e) {
-      logger.log(Level.WARNING, "Exception thrown when registering MetricDescriptor.", e);
-      return false;
+      CreateMetricDescriptorRequest request =
+          CreateMetricDescriptorRequest.newBuilder()
+              .setNameWithProjectName(projectName)
+              .setMetricDescriptor(metricDescriptor)
+              .build();
+      try {
+        metricServiceClient.createMetricDescriptor(request);
+        tracer.getCurrentSpan().addAnnotation("Finish creating MetricDescriptor.");
+        tracer.getCurrentSpan().setStatus(Status.OK);
+        return true;
+      } catch (ApiException e) {
+        logger.log(Level.WARNING, "ApiException thrown when creating MetricDescriptor.", e);
+        tracer
+            .getCurrentSpan()
```

Comment:
You can access directly the span here correct?

Reply:
Right, updated.